src/memray/_memray.pyx

```diff
@@ -4,6 +4,14 @@ import os
 import pathlib
 import sys
 
+
+cdef extern from *:
```

Comment:
Please, add a comment here about why are we doing this. Notice that this should go in the next beta release as I fixed this upstream.

Reply:
> Please, add a comment here about why are we doing this.

Done

> Notice that this should go in the next beta release as I fixed this upstream.

I also added a check on `PY_VERSION_HEX` to capture the fact that this only applies to 3.11 <= version < 3.11.0b4.